File: students/students/src/MON.SPA/src/models/diploma/diplomaAdditionalDocumentModel.js
import Constants from '@/common/constants.js';
import moment from 'moment';

export class DiplomaAdditionalDocumentModel {
  constructor(obj = {}) {
    this.uid = obj.uid || Math.floor((Math.random()) * 0x10000).toString(16);
    this.id = obj.id;
    this.mainDiplomaId = obj.mainDiplomaId;
    this.basicDocumentId = obj.basicDocumentId;
    this.institutionId = obj.institutionId;
    this.institutionName = obj.institutionName;
    this.institutionAddress = obj.institutionAddress;
    this.institutionDetails = obj.institutionDetails;
    this.town = obj.town || null;
    this.municipality = obj.municipality || null;
    this.region = obj.region || null;
    this.localArea = obj.localArea || null;
    this.series = obj.series || null;
    this.factoryNumber = obj.factoryNumber;
    this.registrationNumber = obj.registrationNumber;
    this.registrationNumberYear = obj.registrationNumberYear;
    this.registrationDate = obj.registrationDate ? moment(obj.registrationDate).format(Constants.DATEPICKER_FORMAT) : obj.registrationDate;
  }
}

File: students/students/src/MON.Models/Diploma/DiplomaAdditionalDocumentModel.cs
namespace MON.Models.Diploma
{
    using System;

    public class DiplomaAdditionalDocumentModel
    {
        public int? Id { get; set; }
        /// <summary>
        /// Свързан документ
        /// </summary>
        public int? MainDiplomaId { get; set; }
        public int? BasicDocumentId { get; set; }
        public int? InstitutionId { get; set; }
        public string InstitutionName { get; set; }
        public string InstitutionAddress { get; set; }
        public string Town { get; set; }
        public string Municipality { get; set; }
        public string Region { get; set; }
        public string LocalArea { get; set; }
        public string Series { get; set; }
        public string FactoryNumber { get; set; }
        public string RegistrationNumber { get; set; }
        public string RegistrationNumberYear { get; set; }
        public DateTime? RegistrationDate { get; set; }
    }

    public class DiplomaAdditionalDocumentViewModel : DiplomaAdditionalDocumentModel
    {
        public string BasicDocumentName { get; set; }
        public string InstitutionDetails { get; set; }
    }
}